### ISSUE TITLE
oxfmt: SCSS syntax error on complex selectors inside `:host-context()` — regression in v0.57.0

### ISSUE THREAD

**Issue by Shinigami92** [oxc-project/oxc#24726]
### What happened?

Since oxfmt v0.57.0, `--check` fails with a syntax error on SCSS files that use a complex selector (descendant combinator) inside the `:host-context()` pseudo-class:

```
  x Syntax error: expect token `)`, but found `.`
   ,-[color-circle.component.scss:3:30]
 2 |
 3 | :host-context(pl-color-panel .controls) {
   :                              ^
 4 |   .color-wrap {
   `----
Error occurred when checking code style in the above files.
```

oxfmt v0.56.0 accepted these files (`All matched files use the correct format.`).
The regression matches the v0.57.0 switch from the Prettier fallback to the new `oxc_formatter_css`.

### Minimal reproduction

```scss
/* repro.scss */
:host-context(.parent .child) {
  color: red;
}

:host-context(app-render pl-part) {
  color: blue;
}
```

```sh
pnpm dlx oxfmt@0.56.0 --check repro.scss   # passes
pnpm dlx oxfmt@0.57.0 --check repro.scss   # Syntax error: expect token `)`
pnpm dlx prettier --check repro.scss       # parses fine (style warnings only)
```

### Why this should parse

Per the CSS Scoping spec, `:host-context()` takes a `<compound-selector>`, so the new parser is technically spec-correct. However:

- Prettier (postcss-scss) parses and formats these selectors without error, so this is a Prettier-compatibility difference, and a hard error blocks formatting of the whole file.
- This pattern is idiomatic in Angular: with emulated view encapsulation the Angular compiler (`ShadowCss`) explicitly supports combinators inside `:host-context()` and rewrites them to plain CSS, so real-world Angular codebases use it widely. The selector never reaches the browser as-is.

A formatter should ideally not be stricter than the tool it aims to be compatible with — either parse the argument leniently or fall back to printing the selector verbatim instead of erroring.

### Environment

- oxfmt: 0.57.0 / 0.58.0 (broken), 0.56.0 (ok)
- macOS (darwin arm64), invoked via `oxfmt --check` on an Angular workspace

**Comment by leaysgur** [comment]
NOTE: We should relax `:host`, `:host-context` like others in `oxc-css-parser` side.
https://github.com/oxc-project/oxc-css-parser/pull/130

**Comment by Shinigami92** [comment]
@leaysgur I collected $15 which we sponsored for this to you 💖

**Comment by leaysgur** [comment]
Wow, thank you so much~! 🙌🏻
